lib/features/push_notification/presentation/controller/fcm_message_controller.dart

```diff
@@ -227,76 +213,98 @@ class FcmMessageController extends FcmBaseController {
-    } else {
-      _clearPayloadData();
-      logError('FcmMessageController::_getSessionAction():_getSessionInteractor is null');
+      consumeState(_getSessionInteractor!.execute(stateChange: stateChange));
     }
   }
 
   void _handleGetSessionSuccess(GetSessionSuccess success) {
-    _currentSession = success.session;
-    _userName = success.session.username;
-    final apiUrl = success.session.getQualifiedApiUrl(baseUrl: _dynamicUrlInterceptors?.jmapUrl);
-    log('FcmMessageController::_pushActionFromRemoteMessageBackground():apiUrl: $apiUrl');
-    if (apiUrl.isNotEmpty) {
-      _dynamicUrlInterceptors?.changeBaseUrl(apiUrl);
-      _pushActionFromRemoteMessageBackground();
-    } else {
-      _clearPayloadData();
-      logError('FcmMessageController::_handleGetSessionSuccess():apiUrl is null');
-    }
-  }
-
-  void _pushActionFromRemoteMessageBackground() {
-    log('FcmMessageController::_pushActionFromRemoteMessageBackground():_payloadData: $_payloadData | _currentAccountId: $_currentAccountId | _currentSession: $_currentSession');
-    if (_payloadData?.isNotEmpty == true && _currentAccountId != null && _userName != null) {
-      final stateChange = _parsingPayloadData(_payloadData!);
-      final mapTypeState = stateChange.getMapTypeState(_currentAccountId!);
-      _mappingTypeStateToAction(mapTypeState, _currentAccountId!, _userName!, isForeground: false, session: _currentSession);
+    try {
+      final apiUrl = success.session.getQualifiedApiUrl(baseUrl: _dynamicUrlInterceptors?.jmapUrl);
+      final stateChange = success.stateChange;
+
+      if (apiUrl.isNotEmpty && stateChange != null) {
+        _dynamicUrlInterceptors?.changeBaseUrl(apiUrl);
+
+        _pushActionFromRemoteMessageBackground(
+          accountId: success.session.personalAccount.accountId,
+          userName: success.session.username,
+          stateChange: stateChange,
+          session: success.session);
+      }
+    } catch (e) {
+      logError('FcmMessageController::_handleGetSessionSuccess: Exception $e');
     }
-    _clearPayloadData();
   }
 
-  void _clearPayloadData() {
-    _payloadData = null;
+  void _pushActionFromRemoteMessageBackground({
+    required AccountId accountId,
+    required UserName userName,
+    required StateChange stateChange,
+    Session? session
+  }) {
+    final mapTypeState = stateChange.getMapTypeState(accountId);
+
+    _mappingTypeStateToAction(
+      mapTypeState,
+      accountId,
+      userName,
+      isForeground: false,
+      session: session);
   }
 
   @override
   void handleFailureViewState(Failure failure) {
     log('FcmMessageController::_handleFailureViewState(): $failure');
-    _clearPayloadData();
   }
 
   @override
   void handleSuccessViewState(Success success) {
     log('FcmMessageController::_handleSuccessViewState(): $success');
-    if (success is GetAuthenticatedAccountSuccess) {
-      _handleGetAuthenticatedAccountSuccess(success);
```

Comment:
why don't need to `getAuthenticatedAccount`?

Reply:
We just need to get its value to process on the interactor side so we don't need to check it on the controller side anymore.